packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -108,15 +109,15 @@ export function useGlobalSetting( propertyPath, blockName, source = 'all' ) {
 			);
 		}
 
-		const result = {};
+		let result = {};
 		VALID_SETTINGS.forEach( ( setting ) => {
 			const value =
 				get(
 					configToUse,
 					`settings${ appendedBlockPath }.${ setting }`
 				) ?? get( configToUse, `settings.${ setting }` );
 			if ( value ) {
-				set( result, setting, value );
+				result = setImmutably( result, setting.split( '.' ), value );
```

Comment:
Pretty straightforward to trace - `result` is already declared to be an object, and `setting` is already a string, which is handled just fine by `setImmutably` once we split it to an array path.